types/types.go

```diff
@@ -995,12 +995,18 @@ func GetLHVolumeAttachmentNameFromVolumeName(volName string) string {
 
 // IsSelectorsInTags checks if all the selectors are present in the tags slice.
 // It returns true if all selectors are found, false otherwise.
-func IsSelectorsInTags(tags, selectors []string) bool {
+func IsSelectorsInTags(tags, selectors []string, tolerateEmptySelector bool) bool {
 	if !sort.StringsAreSorted(tags) {
 		logrus.Debug("BUG: Tags are not sorted, sorting now")
 		sort.Strings(tags)
 	}
 
+	if !tolerateEmptySelector {
+		if len(selectors) == 0 && len(tags) != 0 {
+			return false
+		}
+	}
```

Comment:
NIT: I would prefer the following format. Since `len(selectors) == 0` is the only exception of the below code `for _, selector := range selectors`. Listing this special case handling together with the general handling makes the code more easy-to-understand.
```
	if len(selectors) == 0 {
		if !tolerateEmptySelector && len(tags) != 0 {
			return false
		}
	}
```